payload_test.go

```diff
@@ -0,0 +1,90 @@
+package bugsnag
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/bugsnag/bugsnag-go/errors"
+	"github.com/bugsnag/bugsnag-go/sessions"
+)
+
+const expSmall = `{"apiKey":"","events":[{"app":{"releaseStage":""},"device":{},"exceptions":[{"errorClass":"","message":"","stacktrace":null}],"metaData":{},"payloadVersion":"4","severity":"","unhandled":false}],"notifier":{"name":"Bugsnag Go","url":"https://github.com/bugsnag/bugsnag-go","version":"1.4.0"}}`
+
+// The large payload has a timestamp in it which makes it awkward to assert against.
+// Instead, assert that the timestamp property exist, along with the rest of the expected payload
+const expLargePre = `{"apiKey":"166f5ad3590596f9aa8d601ea89af845","events":[{"app":{"releaseStage":"mega-production","type":"gin","version":"1.5.2"},"context":"/api/v2/albums","device":{"hostname":"super.duper.site"},"exceptions":[{"errorClass":"error class","message":"error message goes here","stacktrace":[{"method":"doA","file":"a.go","lineNumber":65},{"method":"fetchB","file":"b.go","lineNumber":99,"inProject":true},{"method":"incrementI","file":"i.go","lineNumber":651}]}],"groupingHash":"custom grouping hash","metaData":{"custom tab":{"my key":"my value"}},"payloadVersion":"4","session":{"startedAt":"`
```

Comment:
I did consider doing that. Small changes in these tests would mean that we also had to change fixture files, which should really live in a subdirectory somewhere to avoid polluting this package. But since it's only used by one test file I decided against doing that. It's easier to see where related code lives this way.